proposed/vcpkg.md

```diff
@@ -0,0 +1,91 @@
+# Using vcpkg in .NET projects and the VMR
+
+**Owner** [Jeremy Koritzinsky](https://github.com/jkoritzinsky)
+
+## Scenarios and User Experience
+
+In .NET today, we handle native dependencies and native dependency flow by a few different mechanisms:
+
+- Vendoring dependencies, aka including a copy of the source with our applied patches in the product repository.
+- Git submodules pointing to the source repo of the dependency.
+- NuGet packages that flow pre-built between our repositories.
+
+For many of our dependencies, these build options work decently well. However, these mechanisms have a few drawbacks:
+
+- Vendoring code increases the size of our repositories and increases build times.
```

Comment:
So far, we have been vendoring fairly small projects. Vendoring does not substantially increase the repo footprint.

Personally, I love the simplicity and flexibility of vendoring for our use cases.

Reply:
dotnet/runtime uses vendoring for its native dependencies today, but dotnet/aspnetcore does not (they use submodules), possibly because their dependencies are more optional (Windows-only as the native code is IIS-specific) and they're larger (GoogleTest is pretty chunky).

Our mechanism for vendoring code also requires us to add additional (although I agree small) CMake logic to support our source-build customers opting-out of our copies of libraries and using the system libraries instead.

I believe we could adjust how we vendor our projects to help mitigate the second issue as well as other issues.

Reply:
Vendoring is not SB friendly in cases where the dependency is already included in a distro.  Distro maintainers would prefer our applied patches were upstreamed.  This reduces the maintenance overhead for them.  From a security perspective there is only one version to maintain.